src/main/java/vendingmachine/model/Menus.java

```diff
@@ -0,0 +1,52 @@
+package vendingmachine.model;
+
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Menus {
+    private final Set<Menu> menus;
+
+    public Menus(Set<Menu> menus) {
+        isValidMenus(menus);
+        this.menus = new HashSet<>(menus);
+    }
+
+    private void isValidMenus(Set<Menu> menus) {
+        Set<String> menuNameSet = menus.stream().map(Menu::getName).collect(Collectors.toSet());
+        if (menus.size() != menuNameSet.size()) {
+            throw new IllegalArgumentException("[ERROR] 동일한 이름의 메뉴가 포함되어 있습니다");
+        }
+    }
+
+    public Menu buyMenu(String menuName) {
+        Menu sale = menus.stream()
+                .filter(menu -> menu.getName().equals(menuName))
```

Comment:
메시지를 던져보는건 어떨까요? 😁

Reply:
앗, 이 부분에서 예외를 던져두고 예외처리 로직에 포함하지 않았네요..!!
프로그램이 바로 종료되는 오류를 방금 경험했습니다..

콘솔 출력으로 바꾸겠습니다!! 꼼꼼하게 봐주셔서 감사합니다!!
사실 코드 리뷰를 부탁드리기 전에 제가 몇 번이나 확인해야 하는 것이 맞는데 부끄럽네요

Reply:
```suggestion
                .filter(menu -> menu.isSameMenuName(menuName))
```

메시지를 던지라는 의미는 이렇게 동작하도록 구현을 변경해보는걸 의미했답니다. 😂

Reply:
아, 그렇군요 ㅎㅎㅎ
덕분에 오류도 찾았습니다!

메세지를 던지는 행위를 처음 경험해보네요! 한번 구현을 변경해보겠습니다!!